js/services/contact_service.js

```diff
@@ -46,6 +45,11 @@ angular.module('contactsApp')
 							});
 						}).then(function (contacts_) {
 							contacts_.map(function (contact) {
+								// Validate some fields
+								if(contact.fix()) {
+									// Can't use this in those nested functions
+									contactService.update(contact);
+								}
 								contacts.put(contact.uid(), contact);
```

Comment:
This `contacts` references the parameter in line 26.
We should rename the cache to something like `contactsCache` to prevent these name collisions.